RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitHeterogeneous.cc

```diff
@@ -242,9 +242,9 @@ void SiPixelRecHitHeterogeneous::run(const edm::Handle<SiPixelClusterCollectionN
     std::sort_heap(ind, ind+ngh,[&](auto a, auto b) { return mrp[a]<mrp[b];});
     uint32_t ic=0;
     auto jnd = [&](int k) { return fc+ind[k]; };
-    assert(ngh==DSViter->size());
+    assert(ngh<=DSViter->size());
     for (auto const & clust : *DSViter) {
-      assert(ic<ngh);
+      if(ic>=ngh) break;
```

Comment:
what happens when we `break` here ?

Reply:
that the clusters between ```ngh and DSViter->size()```
are not processed.

This is supposed to be consistent with the cut already implemented in
_RecoLocalTracker/SiPixelRecHits/plugins/gpuPixelRecHits.h_
which reads
```
    if (threadIdx.x==0 && nclus > MaxClusInModule) {
      printf("WARNING: too many clusters %d in Module %d. Only first %d processed\n", nclus,me,MaxClusInModule);
      // zero charge: do not bother to do it in parallel
      for (auto d=MaxClusInModule; d<nclus; ++d) { chargeh[d]=0; detInd[d]=InvId;}
    }
    nclus = std::min(nclus, MaxClusInModule);
```

Reply:
at the moment the cut in the clusterizer in larger than ```MaxClusInModule``` to account for Clusters below threshold.
One could implement it in _/RecoLocalTracker/SiPixelClusterizer/plugins/gpuClusterChargeCut.h_